rfc/graph-feature-flags-v2.md

```diff
@@ -0,0 +1,118 @@
+---
+title: "Graph Feature Flags: RFC to add feature flags into your Federated GraphQL API"
+author: Jens Neuse
+---
+
+This is an extension to the initial [Graph Feature Flags RFC](./graph-feature-flags.md)
+with the goal to address some shortcomings of the initial proposal and address the feedback.
+
+The problem with the previous approach was that it doesn't scale well with the number of feature flags.
+The previous approach resulted in a matrix multiplication of all feature flags,
+so the complexity of calculating the composability of all feature flags was O(n^2) with n being the number of feature flags.
+
+This RFC extension proposes an alternative approach wich a complexity of O(n+m) with n being the number of feature flags,
+and m being the number of feature flag groups.
+
+In addition, I'd like to propose a cleaner way to define and maintain feature flags,
+separate from publishing subgraph schemas.
+
+## Solving the Matrix Multiplication Problem
+
+First, we start by publishing a base Subgraph schema:
+
+```bash
+npx wgc subgraph publish products --schema ./products.graphql --routing-url http://localhost:4001/graphql
+```
+
+Then, we publish a feature flag schema for the same subgraph:
+
+```bash
+wgc ff publish myFlag --routing-url http://localhost:4002/graphql --schema ./productsV2.graphql --mode replace --subgraph products --supergraph myGraph
```

Comment:
are the validation flags mentioned in v1 still available? (`--skip-composition-checks` & `--disable-composition-checks`)

Reply:
Well since you said your expectation that a flag would compose maybe they aren't needed? I was going based off the v1 stating those CLI flags were available.